### PR TITLE
[Feature] Upgrade SMPLify and body models

### PR DESCRIPTION
**Pull request by caizhongang**
- [x] Support joint mask in SMPLify and SMPL
- [x] Support body_model (not config) as the input
- [x] Support on-the-flight batch size adaptation
- [x] Adapt to MMHuman3D camera module
- [x] Add unit tests

### CODE REVIEW

File: tests/test_cameras/test_cameras.py [hunk 1]
```diff
@@ -163,3 +163,59 @@ def test_perspective():
     cam2.to_screen_()
     cam2.to_ndc_()
     check_camera_close(cam, cam2)
+
+
+def test_perspective_projection():
```
Comment (caizhongang) [inline tests/test_cameras/test_cameras.py:168]: Add an perspective projection unit test but unable to pass, please help take a look @JasonBoy1

File: configs/smplify/smplify.py [hunk 2]
```diff
@@ -63,3 +63,11 @@
     'neck_openpose', 'right_hip_openpose', 'left_hip_openpose',
     'right_hip_extra', 'left_hip_extra'
 ]
+
```
[the remark this answers is not in the document]
Comment (caizhongang) [reply configs/smplify/smplify.py:66]: As discussed offline, let's use the actual class name wherever possible.

File: configs/smplify/smplifyx.py [hunk 3]
```diff
@@ -87,3 +87,11 @@
     use_full_body=True)
 
 smooth_loss = dict(type='SmoothJointLoss', loss_weight=0, reduction='sum')
+
+camera = dict(
```
[the remark this answers is not in the document]
Comment (caizhongang) [reply configs/smplify/smplifyx.py:91]: As discussed offline, let's use the actual class name wherever possible.

File: configs/spin/resnet50_spin_pw3d.py [hunk 4]
```diff
@@ -45,7 +47,14 @@
     ignore_keypoints=[
         'neck_openpose', 'right_hip_openpose', 'left_hip_openpose',
         'right_hip_extra', 'left_hip_extra'
-    ])
+    ],
+    camera=dict(
```
[the remark this answers is not in the document]
Comment (caizhongang) [reply configs/spin/resnet50_spin_pw3d.py:51]: As discussed offline, let's use the actual class name wherever possible.

File: tests/test_cameras/test_cameras.py [hunk 5]
```diff
@@ -163,3 +163,55 @@ def test_perspective():
...
+        # Apply camera intrinsics
+        projected_points = torch.einsum('bij,bkj->bki', K, projected_points)
+
+        return projected_points[:, :, :-1]
+
+    focal_length = 5000
+    image_size = (224, 224)
+    principal_point = (112, 112)
+
+    camera = build_cameras(
```
[the remark this answers is not in the document]
Comment (caizhongang) [reply tests/test_cameras/test_cameras.py:195]: As discussed offline, let's use the actual class name wherever possible.

File: tests/test_models/test_architectures/test_mesh_estimator.py [hunk 6]
```diff
@@ -316,6 +316,15 @@ def test_run_registration():
         extra_joints_regressor='data/body_models/J_regressor_extra.npy',
         batch_size=batch_size)
 
+    camera = build_cameras(
```
[the remark this answers is not in the document]
Comment (caizhongang) [reply tests/test_models/test_architectures/test_mesh_estimator.py:319]: As discussed offline, let's use the actual class name wherever possible.

File: tests/test_smplify.py [hunk 7]
```diff
@@ -9,29 +9,39 @@
...
+        batch_size=1  # need not to know batch size at init
+    )
     smplify_config['num_epochs'] = 1
     smplify_config['use_one_betas_per_video'] = True
 
     smplify = build_registrant(smplify_config)
 
     # Generate keypoints
-    smpl = build_body_model(body_model_config)
+    smpl = build_body_model(
```
[the remark this answers is not in the document]
Comment (caizhongang) [reply tests/test_smplify.py:35]: As discussed offline, let's use the actual class name wherever possible.

**Review by JasonBoy1 — APPROVED** [review]
